docs/gen-ai/llm-metrics.md

```diff
@@ -0,0 +1,114 @@
+| [`gen_ai.usage.token_type`](../attributes-registry/llm.md) | string | The type of token being counted. | `prompt`; `completion` | `Required` | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| [`server.address`](../attributes-registry/server.md) | string | Server domain name if available without reverse DNS lookup; otherwise, IP address or Unix domain socket name. [1] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | `Required` | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+| [`error.type`](../attributes-registry/error.md) | string | Describes a class of error the operation ended with. [2] | `timeout`; `java.net.UnknownHostException`; `server_certificate_invalid`; `500` | `Conditionally Required` if the operation ended in an error | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+
+**[1]:** When observed from the client side, and when communicating through an intermediary, `server.address` SHOULD represent the server address behind any intermediaries, for example proxies, if it's available.
+
+**[2]:** The `error.type` SHOULD be predictable and SHOULD have low cardinality.
+Instrumentations SHOULD document the list of errors they report.
+
+The cardinality of `error.type` within one instrumentation library SHOULD be low.
+Telemetry consumers that aggregate data from multiple instrumentation libraries and applications
+should be prepared for `error.type` to have high cardinality at query time when no
+additional filters are applied.
+
+If the operation has completed successfully, instrumentations SHOULD NOT set `error.type`.
+
+If a specific domain defines its own set of error identifiers (such as HTTP or gRPC status codes),
+it's RECOMMENDED to:
+
+* Use a domain-specific attribute
+* Set `error.type` to capture all errors, regardless of whether they are defined within the domain-specific set or not.
+
+`error.type` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
+
+| Value  | Description | Stability |
+|---|---|---|
+| `_OTHER` | A fallback error value to be used when the instrumentation doesn't define a custom value. | ![Stable](https://img.shields.io/badge/-stable-lightgreen) |
+<!-- endsemconv -->
+
+### Metrics: `gen_ai.request.duration`
+
+This metric is [required][MetricRequired].
+
+<!-- semconv metric.gen_ai.request.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `gen_ai.request.duration` | Histogram | `s` | LLM request duration | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+<!-- semconv metric.gen_ai.request.duration(full) -->
+| Attribute  | Type | Description  | Examples  | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) | Stability |
```

Comment:
I wonder if we should have some notion of "type" here as an attribute, although we still haven't defined that for `llm-spans`. For example, we should differentiate between calls to get the embeddings of texts which are usually fast and calls to get chat completions). They should have different response model parameters ofc, but that's another way to slice this metric IMO.

Reply:
agree, we need `gen_ai.operation.name` or something similar

Reply:
Should we reuse the `gen_ai.request` namespace for this? And is `type` or `name` a better convention?

Seems each area has a different convention:

HTTP: http.request.method
DNS: dns.question.name
RPC: rpc.method
Messaging: messaging.operation.type

Reply:
Messaging and databases have `*.operation,name` since a few days ago, but it's rather a pattern and not a requirement we have to follow.

Comparing request vs operation, the request is more of a physical thing (like HTTP request), but gen_ai semconv apply to logical level where operation name works better.

E.g. if there are multiple reties on the HTTP side, or if there is an auth request to authenticate HTTP call, etc, we'd still have one gen_ai operation. 

WRT type vs name - both would work, but name is broader and more extensible (https://github.com/open-telemetry/semantic-conventions/issues/947)

Reply:
I've added gen_ai.operation.name for now.